pkgs/stdenv/darwin/default.nix

```diff
@@ -98,7 +101,10 @@ in rec {
       };
 
     in {
-      inherit system platform crossSystem config;
+      buildPlatform = localSystem;
+      hostPlatform = localSystem;
+      targetPlatform = localSystem;
```

Comment:
Nope. Darwin itself is never cross compiling, see the assert at the top. When `stdenv/cross/default.nix` recurs to solve for the native bootstrap with which to extend, it should take care of this.